libc/newhdrgen/tests/input/test_string.yaml

```diff
@@ -0,0 +1,303 @@
+header: string.h
```

Comment:
for the test, you should make a much smaller yaml file, probably with a different name as well (so it's clear this is just for testing). You also need to make sure you're testing all of the parts, not just functioins. I'd recommend something like this:

```
header: test_header.h
macros: [MACRO_A, MACRO_B]
types: [type_a, type_b]
enums: [enum_a: [value_1, value_2], enum_b: [value_3]]
objects: [object_1, object_2]
functions: [list of functions]
```